cli/cmd/packageInfo.go

```diff
@@ -0,0 +1,248 @@
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			path := args[0]
+
+			// Check if given filename is actually a directory
+			stat, err := os.Stat(path)
+			if err != nil {
+				return err
+			}
+			isDirectory := stat.IsDir()
+
+			// For Open Enclave / Edgeless RT / EGo, we require to directly point to the signed enclave binary, as these do not have a specific directory structure
+			var errOpenEnclave error
+			if !isDirectory {
+				errOpenEnclave = decodeOpenEnclaveSigStruct(path)
+			}
+
+			// In every other case, try to guess if it's a directory, or expect a specific file to be pointed to
+			errGraphene := decodeGrapheneSigStruct(path, isDirectory)
+			errOcclum := decodeSGXSDKSigStruct(path, isDirectory) // Either Occlum or SGX SDK
+
+			color.Red("ERROR: Failed to automatically determine SGX package signature properties detection.")
+			if isDirectory {
+				color.Red("A directory was supplied, but it appears not to be a Graphene or Occlum instance.")
+				color.Red("Please either specify the .sig file (Graphene) or SGX enclave binary (Occlum / SGX SDK) directly, or the root directory of an Graphene or Occlum instance.")
+			}
+
+			fmt.Printf("\n")
+			// Output exact errors for each detection method to user
+			if errOpenEnclave != nil {
+				color.Red("Open Enclave detection error: %v\n", errOpenEnclave)
+			}
+			fmt.Printf("Error - Graphene: %v\n", errGraphene)
+			fmt.Printf("Error - Occlum / SGX SDK: %v\n", errOcclum)
+			return errors.New("unable to determine enclave properties")
+		},
+		SilenceUsage: true,
+	}
+
+	return cmd
+}
+
+func decodeSGXSDKSigStruct(path string, isDirectory bool) error {
+	// If the path is a directory, we try to find out if it's an Occlum image directory
+	var elfFile *elf.File
+	var isOcclumInstance bool
+	var err error
+	if isDirectory {
+		if elfFile, err = elf.Open(filepath.Join(path, "build/lib/libocclum-libos.signed.so")); err == nil {
+			color.Green("Detected Occlum image.")
+			isOcclumInstance = true
+		} else if os.IsNotExist(err) {
+			return err
+		}
+	} else {
+		elfFile, err = elf.Open(path)
+	}
+	if err != nil {
+		return err
+	}
+	defer elfFile.Close()
+
+	// SIGSTRUCT is stored in ELF section '.note.sgxmeta'
+	sgxMetaSection := elfFile.Section(".note.sgxmeta")
+	if sgxMetaSection == nil {
+		return errors.New("could not find SGX metadata section (.note.sgxmeta) in given file")
+	}
+
+	sgxMetaData, err := sgxMetaSection.Data()
+	if err != nil {
+		return err
+	}
+
+	// Parse and retrieve values from SIGSTRUCT
+	mrenclave, mrsigner, isvprodid, isvsvn, err := parseSigStruct(sgxMetaData)
+	if err != nil {
+		return nil
+	}
+
+	// Display the determined properties
+	if isOcclumInstance {
+		color.Cyan("PackageProperties for Occlum image at '%s':\n", path)
+	} else {
+		color.Cyan("PackageProperties for '%s':\n", path)
+	}
+
+	printPackageProperties(mrenclave, mrsigner, isvprodid, isvsvn)
+
+	os.Exit(0)
```

Comment:
```suggestion
```
I'd prefer to let the caller check the error instead. It's a bit more LOC, but the caller's control flow is currently unobvious.